test/system_test.py

```diff
@@ -650,6 +637,8 @@ def test_install_default(self):
         spack_install_and_test('py-hatchling')
 
 
+@pytest.mark.no_daint  # problem with gt4py and spack v21.1
+@pytest.mark.no_balfrin  # problem with gt4py and spack v21.1
```

Comment:
This implies that icon4py is not tested any more. Thus the tests are obsolete. Please remove them.

Reply:
@abishekg7 @samkellerhals This means no more tests for icon4py. Is this okay?

Reply:
Apparently py-icon4py doesn't work on Daint, Tsa, Balfrin. I think this makes the package useless. Consider removing it.

Reply:
EXCLAIM is aware of the issues with v0.21.1 since over a month.
I talked with Anurag and Abishek. They both know about this degradation and the responsible people at gt4py-team are aware of it.

But for reasons I don't understand EXCLAIM won't fix this unless their pipelines are broken.
Only then they will allocate resources to fix it.

Since C2SM as a whole needs to prepare for Alps and the software stacks built on top of v0.21.1 we all (Abishek, Anurag and me) agreed to do exactly what this PR is about to do.

Reply:
Since this is a known defect, I think we should do one of the following things for now:
- Remove the icon4py package and the tests.
- Add a warning to the README.md, like "Warning: icon4py is broken"
- Create an issue for it.
It's only fair to propagate this information, otherwise someone might accidently waist time on it.

We can reintroduce icon4py once it's fixed.